ARCs/arc-0018.md

```diff
@@ -0,0 +1,449 @@
+
+### Royalty Policy 
+
+A Royalty Share consists of an `address` that should receive a Royalty payment and a `share` representing some share of the total payment amount. The Royalty Share **MUST** be specified in terms of [basis points](https://en.wikipedia.org/wiki/Basis_point) of the payment amount.
+
+A Royalty Policy is specified as a Royalty Collector and associated Royalty Share. Additionally, if any assets besides Algos are acceptable, 
+
+
+```ts
+interface RoyaltyPolicy {
+    share:              number,  // The share of the payment due, specified in basis points (0-10,000)
+    royalty_collector:  string,  // The address that should collect this share of the payment
+    allowed_assets:     number[] // The list of asset ids that are acceptable for payment
+}
+```
+A given asset has at most one associated policy.
+
+> In case royalty needs to be distributed to multiple different shareholders, the Royalty Collector address may be set to an appropriate Smart Signature or Application Account of a Smart Contract that would ensure this distribution.
+### Royalty Enforcer
+
+The Royalty Enforcer is an application that controls the transfer of ASAs subject to the Royalty Policy. 
+
+This is accomplished by exposing an interface defined as a set of [ABI Methods](#abi-methods) allowing a grouped transaction call.
+
+### Royalty Collector
+
+The Royalty Collector is a generic account that could be set to a Single Signature, a Multi Signature, a Smart Signature or even to another Smart Contract. The Royalty Collector is then responsible for the royalty distribution logic, making the Royalty Enforcement Specification more general and composable.
+
+
+### Royalty Asset 
+
+The Royalty Asset is an ASA subject to royalty collection and **MUST** be created with the the appropriate mutable and immutable parameters. 
+
+> Because the protocol does not allow updating an address parameter after it's been deleted, if the creator thinks they may want to modify them later, they must be set to some non-zero address.
+
+
+#### Asset Address Parameters 
+
+The Clawback parameter **MUST** be set to the Application Address of the Royalty Enforcer.
+
+> Since the Royalty Enforcer relies on using the Clawback mechanism to perform the transfer the Clawback should NEVER be set to the zero address. 
+
+The Freeze parameter **MUST** be set to the Application Address of the Royalty Enforcer if `FreezeAddr != ZeroAddress`, else set to `ZeroAddress`.
+
+> If the creator wants to allow an ASA to be Royalty Free after some conditions are met, it should be set to the Application Address 
+
+The Manager parameter **MUST** be set to the Application Address of the Royalty Enforcer if `ManagerAddr != ZeroAddress`, else set to `ZeroAddress`.
+
+> If the creator wants to update the Freeze parameter, this should be set to the application address
```

Comment:
What happens if a marketplace needs to burn an NFT? If the manager address must be the Application Address, then the Royalty Enforcer needs to be able to burn the NFT as well, or I guess that could be an optional feature.

Reply:
Yes this standard does not mandate what you could do as Manager through the Application, this is a free choice of the implementation (e.g. could let Royalty ASA "admins" act like a Manager using the Application as a proxy).